Tests/SourceKitLSPTests/DocumentTestDiscoveryTests.swift

```diff
@@ -210,14 +210,14 @@ final class DocumentTestDiscoveryTests: XCTestCase {
       tests,
       [
         TestItem(
-          id: "MyTests",
+          id: "MyTests/test.swift:2:0",
```

Comment:
I just checked and test suites aren’t file- and position-qualified. If you run `swift test --filter 'MyTests/test.swift:2:0'` in this project, it wouldn’t execute any tests.

Reply:
Correct, because fully qualified type names are not ambiguous. Function names however can be ambiguous:

```swift
@Suite MyTests {
  @Test(arguments: 0 ... 100)
  func example(i: Int) { ... }

  @Test(arguments: ["a", "b", "c"])
  func example(i: Character) { ... }
}
```

The ID of a test function thus includes its fully-qualified containing type name (if applicable), the name of the function including labels (`example(i:)` in the above case), and a source location.

For convenience, the source location is not required to be specified with `swift test --filter`—if two tests have the same name, we'll just run 'em both in that case. And when you run `swift test list`, we don't show source locations for unambiguous test function names unless you pass `--verbose`.

Reply:
Well, as of https://github.com/apple/swift-testing/pull/589 anyway.